### PR TITLE
openwisp-config: fix Makefile for 1.1.0

### LINKED ISSUE #25168: openwisp-config: instance1 s in a crash loop x
Maintainer: @pandafy, @nemesifier 

**Environment:** mediatek/filogic, Cudy WR3000 v1, OpenWrt 23.05.5 r24106-10cc5fcd00

But likely impacting all users of package **openwisp-config version 1.1.0-1**

**Description:**
Upgraded the package of openwisp-config to version 1.1.0-1 in OpenWrt 23.05.5. I couldn't connect to the controller anymore.

In logs this message appears (and controller doesn't see the client):

`Instance openwisp_config::instance1 s in a crash loop 6`

**Also see:**

[https://github.com/openwisp/openwisp-config/issues/223](https://github.com/openwisp/openwisp-config/issues/223)

Hunting down the bug I discovered that /etc/init.d/openwisp_config wants to execute PROG="/usr/sbin/openwisp-config" but it does not exists, it is called PROG="/usr/sbin/openwisp_config" according to the changelog there were changes re: the - 'hyphen' vs _ 'underscore' ; maybe it just gets packaged incorrectly? Once I edited /etc/init.d/openwisp_config and changed the - 'hyphen' to a _ 'underscore' in PROG= the new agent also started to work. Easier than rolling back.

Looks like the problem is in the OpenWrt Makefile - https://github.com/openwrt/packages/blob/dd2e7105e6f0abf9b72155e2b26298ed19a92ab9/admin/openwisp-config/Makefile which needs to be updated to produce hyphens instead of underscores.

```
	$(INSTALL_BIN) \
		$(PKG_BUILD_DIR)/openwisp-config/files/openwisp.agent \
		$(1)/usr/sbin/openwisp_config

	$(INSTALL_BIN) \
		$(PKG_BUILD_DIR)/openwisp-config/files/openwisp.init \
		$(1)/etc/init.d/openwisp_config
```

should maybe be

```
	$(INSTALL_BIN) \
		$(PKG_BUILD_DIR)/openwisp-config/files/openwisp.agent \
		$(1)/usr/sbin/openwisp-config

	$(INSTALL_BIN) \
		$(PKG_BUILD_DIR)/openwisp-config/files/openwisp.init \
		$(1)/etc/init.d/openwisp-config

```

### PR DESCRIPTION
**Pull request by pandafy**
Update configuration in Makefile to fix #25168

Maintainer: @nemesifier 
Compile tested: ramips
Run tested: ramips

### CODE REVIEW

**Review by nemesifier — APPROVED** [review]
LGTM :heavy_check_mark:

**Review by nemesifier — COMMENTED** [review]
@feckert we fixed some issues and updated some outdated code here.

**Comment by nemesifier** [comment]
Any maintainer could merge this please? @BKPepe

File: admin/openwisp-config/Makefile [hunk 1]
```diff
@@ -38,23 +38,24 @@ endef
...
 
 	$(INSTALL_BIN) \
 		$(PKG_BUILD_DIR)/openwisp-config/files/openwisp.agent \
-		$(1)/usr/sbin/openwisp_config
+		$(1)/usr/sbin/openwisp-config
 
 	$(INSTALL_BIN) \
 		$(PKG_BUILD_DIR)/openwisp-config/files/openwisp.init \
-		$(1)/etc/init.d/openwisp_config
+		$(1)/etc/init.d/openwisp-config
```
Comment (BKPepe) [inline admin/openwisp-config/Makefile:58]: While checking the commit description, this one is only valid, isnt it? :)

File: admin/openwisp-config/Makefile [hunk 2]
```diff
@@ -95,7 +96,24 @@ define Package/openwisp-config/install
...
+#!/bin/sh
+if [ ! -L $${IPKG_INSTROOT}/usr/sbin/openwisp_config ]; then
+	ln -s /usr/sbin/openwisp-config $${IPKG_INSTROOT}/usr/sbin/openwisp_config
+fi
+endef
+
+define Package/openwisp-config/postrm
+#!/bin/sh
+rm -f $${IPKG_INSTROOT}/usr/sbin/openwisp_config
 endef
```
Comment (BKPepe) [inline admin/openwisp-config/Makefile:117]: This will be removed automatically. No needed.

File: admin/openwisp-config/Makefile [hunk 3]
```diff
@@ -95,7 +96,24 @@ define Package/openwisp-config/install
...
+	$(CP) $(PKG_BUILD_DIR)/VERSION $(1)/usr/lib/openwisp-config/
+endef
+
+# for backward compatibility
+define Package/openwisp-config/postinst
+#!/bin/sh
+if [ ! -L $${IPKG_INSTROOT}/usr/sbin/openwisp_config ]; then
+	ln -s /usr/sbin/openwisp-config $${IPKG_INSTROOT}/usr/sbin/openwisp_config
+fi
+endef
```
Comment (BKPepe) [inline admin/openwisp-config/Makefile:112]: Also, this one, well, its breaking change introduced by previous version. Thus users will run this package by init script, right? I find this useless.

File: admin/openwisp-config/Makefile (hunk 2 above)
Comment (nemesifier) [reply admin/openwisp-config/Makefile:117]: The agent is now named `/usr/sbin/openwisp-config`, we create a symbolic link to `/usr/sbin/openwisp_config` for backward compatibility. This postrm hook removes the symbolic link. Was this clear?

File: admin/openwisp-config/Makefile (hunk 3 above)
Comment (nemesifier) [reply admin/openwisp-config/Makefile:112]: Ok I guess we can remove it. Initially we copied the init script too but it caused to have multiple agents running.

File: admin/openwisp-config/Makefile (hunk 2 above)
Comment (BKPepe) [reply admin/openwisp-config/Makefile:117]: I know what you would like to achieve, but I am asking why do it in such complicated way.

File: admin/openwisp-config/Makefile [hunk 4]
```diff
@@ -95,7 +96,24 @@ define Package/openwisp-config/install
 		$(PKG_BUILD_DIR)/openwisp-config/files/sbin/openwisp-get-address.lua \
 		$(1)/usr/sbin/openwisp-get-address
 
-	$(CP) $(PKG_BUILD_DIR)/VERSION $(1)/etc/openwisp/
+	$(INSTALL_BIN) \
+		$(PKG_BUILD_DIR)/openwisp-config/files/sbin/openwisp-get-random-number.lua \
+		$(1)/usr/sbin/openwisp-get-random-number
+
+	$(CP) $(PKG_BUILD_DIR)/VERSION $(1)/usr/lib/openwisp-config/
```
[the remark this answers is not in the document]
Comment (nemesifier) [reply admin/openwisp-config/Makefile:103]: it's used by the program to read the version number

[the remark this answers is not in the document]
Comment (nemesifier) [reply admin/openwisp-config/Makefile:103]: I did not understand this commit.

**Review by nemesifier — COMMENTED** [review]
Ready for review.

**Review by nemesifier — COMMENTED** [review]
Any blocker here?

**Comment by feckert** [comment]
@BKPepe Do you have any comments or can I merge this?

**Comment by BKPepe** [comment]
Yep, I do. :) There is missing PKG_RELEASE in the Makefile.

File: admin/openwisp-config/Makefile [hunk 5]
```diff
@@ -38,23 +38,24 @@ endef
 
 define Package/openwisp-config/conffiles
 /etc/config/openwisp
+/etc/openwisp/
```
Comment (BKPepe) [inline admin/openwisp-config/Makefile:41]: Would be great to mention this additional thing in the commit description.

Comment (nemesifier) [reply admin/openwisp-config/Makefile:41]: It was added by @pandafy, is it ok? Any other blocker?

**Comment by feckert** [comment]
I would also suggest to bump the PKG_RLEASE by one. Everything else looks good to me.

File: admin/openwisp-config/Makefile [hunk 6]
```diff
@@ -6,7 +6,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=openwisp-config
 PKG_VERSION:=1.1.0
-PKG_RELEASE:=1
+PKG_RELEASE:=2
```
Comment (pandafy) [inline admin/openwisp-config/Makefile:9]: @feckert I have already bumped it here.

Comment (feckert) [reply admin/openwisp-config/Makefile:9]: Sorry, I must have overstitched that

**Comment by feckert** [comment]
Thanks merged!

**Pull request merged**